### ISSUE TITLE
Trakt Integration

### ISSUE THREAD

**Issue by drinfernoo** [doko-desuka/plugin.video.toonmania2#7]
Would it be possible to add Trakt integration and lists?

**Comment by doko-desuka** [comment]
Hi. I don't know.
What would you get out of doing that? How do you imagine people should use it? (like what menus, labels etc.)

**Comment by jewbmx** [comment]
Think hes wanting to save shows to his trakt and be able to load it later too. The standard app based list setup might be better tho.

**Comment by doko-desuka** [comment]
@jewbmx Thanks. Maybe they're after being able to add shows on a computer and play them in Toonmania2 in Kodi.  
@drinfernoo If you have an idea of the Trakt features you want, please list them out.

**Comment by jewbmx** [comment]
I just seen some trakt in a toon addon yesterday think it was the Mast.
Redux one

**Comment by drinfernoo** [comment]
@doko-desuka I want to be able to load up a Trakt list of cartoons we watch, and watch them through this add-on, rather than search for each show individually. I usually add a widget on my homescreen for this.

**Comment by drinfernoo** [comment]
@jewbmx Masterani doesn't have Trakt lists, as far as I know. Either way, it has the option to authorize Trakt in the settings, but it seems to be broken.

**Comment by jewbmx** [comment]
Yea i got the addon but havent tried it past the settings screen lol.

**Comment by doko-desuka** [comment]
@drinfernoo alright, I'll take a look.

@jewbmx Toonmania2 uses the JSON api version of those websites already. It's my biggest add-on so far, you should give it a try.

**Comment by jewbmx** [comment]
Oh I already been using your addons since we first started talking on
github lol. I always find myself bouncing through a few different addons
with the shows I watch tho. Some episodes or seasons issues lol

**Comment by doko-desuka** [comment]
Oh, that's cool!  
For cartoons at least, I think people should use specific add-ons for that. If we included one or more scrapers for websites that specialize in cartoons on something like Yoda or ExodusRedux, those cartoon websites would end up being bombarded with mostly failed web requests, since the people using these Exodus-forks are looking for adult TV series and movies most of the time, not cartoons.  
So like 99% of the time the cartoon websites would spend bandwidth answering to requests for content that they'd never have.

In other words, to help those cartoon websites avoid being massively hit, be careful with who you share those exodus cartoon scrapers with x)

**Comment by jewbmx** [comment]
My addon has this in it and seems to be working properly too...

self.genre_filter = ['animation', 'anime']

**Comment by doko-desuka** [comment]
That's brilliant, I forgot about the genre filter. I'm relieved!

**Comment by jewbmx** [comment]
I forgot all about adding it to the copy i posted for ya guys lol but just
tossed it in a minute ago. Found it in the old gogoanime scraper

**Comment by doko-desuka** [comment]
@drinfernoo I have seriously taken a look at the Trakt API docs, the integration code in Incursion and other forks, but I realized that it'd take more time than I can expend at the moment.

**Comment by drinfernoo** [comment]
@doko-desuka, totally understandable. A man can dream... 😂

**Comment by doko-desuka** [comment]
@drinfernoo I re-read your request, "to load up a Trakt list of cartoons we watch". That's actually not that complicated, the difficult stuff is syncing (adding to lists, creating new lists etc.) and "scrobbling" (changing the status of a Trakt item to watched/unwatched).  

I uploaded version 0.5.0, it added a new search method.  
You need to go to Animetoon/Animeplus > Search > Search by Trakt Lists, then pair the add-on to your account through https://trakt.tv/activate.  
At this moment it only shows the "Lists" you have in your account, excluding the watchlist which is a different thing. So you need to have one or more lists in your account with items in them, then you can access them from this add-on.  
When you select an item it uses the name search, as if you had gone to "Search by Name" and typed that name yourself. It's the only way to do it, the source websites don't have any Trakt integration.
Then you'll see your search results, if there's any, and can play / favourite them.

I'm closing this, feel free to create another Issue if you find any problems.

**Comment by drinfernoo** [comment]
That appears to be something that will work for me! It's a little different from the way I typically browse shows, but it should work just fine :)

You're officially the first cartoon add-on I've seen with Trakt list integration 👍👍👍

**Comment by doko-desuka** [comment]
Cool, I hope it works.  
If you think of any tweaks that would make it more convenient to use, let me know.

**Comment by doko-desuka** [comment]
Sure.  
The reason I made it that way was so that I could favorite multiple search results at once (that single item with the yellow name that shows as soon as the search is complete).  
Some name searches like "naruto" come up with several results, so you can get back to all those results later by favoriting that parent item, in this case called "Naruto Search Results".

I added a new setting called "Show Search Results Group" (default is **enabled**) that shows that parent item. If you disable it you'll get the behavior you're after -- if I understood you right it's Search -> Search by Trakt List -> Pick list & item -> See results.
